tool/site/client/src/pages/withdrawPage.tsx

```diff
@@ -27,36 +27,50 @@ const WithdrawPage: React.FC<WithdrawPageProps> = ({
 }) => {
   const {baseUrl} = useContext(AppContext)
   const [withdrawFormValues, setWithdrawFormValues] = useState<WithdrawFormValues>()
-  if (!withdrawFormValues) {
-    return (
-      <>
-        <Label isPageHeading>Withdraw prescription</Label>
-        <WithdrawForm prescriptionId={prescriptionId} onSubmit={setWithdrawFormValues}/>
-      </>
-    )
-  }
-  const sendWithdrawTask = () => sendWithdraw(baseUrl, prescriptionId, withdrawFormValues)
+
+  const retrieveDispenseNotificationsTask = () => retrieveDispenseNotifications(baseUrl, prescriptionId)
   return (
-    <LongRunningTask<ApiResult> task={sendWithdrawTask} loadingMessage="Sending withdraw.">
-      {withdrawResult => (
-        <>
-          <Label isPageHeading>Withdraw Result {withdrawResult.success ? <TickIcon /> : <CrossIcon />}</Label>
-          <PrescriptionActions prescriptionId={prescriptionId} dispense view/>
-          <MessageExpanders
-            fhirRequest={withdrawResult.request}
-            hl7V3Request={withdrawResult.request_xml}
-            fhirResponse={withdrawResult.response}
-            hl7V3Response={withdrawResult.response_xml}
-          />
-          <ButtonList>
-            <ReloadButton />
-          </ButtonList>
-        </>
-      )}
+    <LongRunningTask<{[key: number]: fhir.Bundle}> task={retrieveDispenseNotificationsTask} loadingMessage="Retrieving dispense notifications.">
+      {dispenseNotifications => {
+        if (!withdrawFormValues) {
+          return (
+            <>
+              <Label isPageHeading>Withdraw prescription</Label>
+              <WithdrawForm dispenseNotifications={Object.values(dispenseNotifications)} prescriptionId={prescriptionId} onSubmit={setWithdrawFormValues}/>
+            </>
+          )
+        }
+        const sendWithdrawTask = () => sendWithdraw(baseUrl, prescriptionId, withdrawFormValues)
+        return (
+          <LongRunningTask<ApiResult> task={sendWithdrawTask} loadingMessage="Sending withdraw.">
+            {withdrawResult => (
+              <>
+                <Label isPageHeading>Withdraw Result {withdrawResult.success ? <TickIcon /> : <CrossIcon />}</Label>
+                <PrescriptionActions prescriptionId={prescriptionId} dispense view/>
+                <MessageExpanders
+                  fhirRequest={withdrawResult.request}
+                  hl7V3Request={withdrawResult.request_xml}
+                  fhirResponse={withdrawResult.response}
+                  hl7V3Response={withdrawResult.response_xml}
+                />
+                <ButtonList>
+                  <ReloadButton />
+                </ButtonList>
+              </>
+            )}
+          </LongRunningTask>
+        )
+      }}
     </LongRunningTask>
   )
 }
 
+async function retrieveDispenseNotifications(baseUrl: string, prescriptionId: string): Promise<{[key: number]: fhir.Bundle}> {
```

Comment:
change the type to `Promise<{dispenseNotifications: Array<fhirBundle>}>`

Reply:
Done